evalml/data_checks/no_variance_data_check.py

```diff
@@ -153,7 +155,10 @@ def validate(self, X, y):
         messages = []
 
         X = infer_feature_types(X)
-        y = infer_feature_types(y)
+        if y is not None:
+            y = infer_feature_types(y)
+        else:
+            y = infer_feature_types(pd.Series([1, 2]))
```

Comment:
Is this intentional?

Reply:
Rather than adding an `if` below and having to indent all the target-related logic (or returning early), I figured I would just use a dummy target that is not "no variance"

Reply:
Hmm, I don't think checking below and then returning early for the target-related logic is too bad of an idea but I'll leave it up to you guys! 

I personally would be more okay hardcoding if this were in the tests and not in the implementation :)